forms-flow-api/src/formsflow_api/resources/dashboards.py

```diff
@@ -28,23 +28,32 @@ class DashboardList(Resource):
     @profiletime
     def get():
         """List all dashboards."""
-        if request.args:
-            dict_data = ApplicationListReqSchema().load(request.args)
-            page_no = dict_data["page_no"]
-            limit = dict_data["limit"]
-        else:
-            page_no = None
-            limit = None
-        response = analytics_service.get_request(
-            url_path="dashboards", page_no=page_no, limit=limit
-        )
-        if response == "unauthorized":
-            return {"message": "Dashboard not found"}, HTTPStatus.NOT_FOUND
-        if response is None:
-            return {"message": "Error"}, HTTPStatus.SERVICE_UNAVAILABLE
+        try:
+            if request.args:
+                dict_data = ApplicationListReqSchema().load(request.args)
+                page_no = dict_data["page_no"]
+                limit = dict_data["limit"]
+            else:
+                page_no = None
+                limit = None
+            response = analytics_service.get_request(
+                url_path="dashboards", page_no=page_no, limit=limit
+            )
+            if response == "unauthorized":
+                return {"message": "Dashboard not found"}, HTTPStatus.NOT_FOUND
```

Comment:
can we use `HTTPStatus.UNAUTHORIZED` here? if so add a related clear message too.

Reply:
resolved